File: 剑指offer/6.从尾到头打印链表.py
from typing import List


class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
        stack = []
        while head:
            stack.append(head)
            head = head.next

        ans = []
        while stack:
            node = stack.pop()
            ans.append(node.val)

        return ans

File: 剑指offer/6.从尾到头打印链表.cpp
#include <vector>
using namespace std;

struct ListNode {
    int val;
    ListNode* next;
    ListNode(int x)
        : val(x)
        , next(NULL)
    {
    }
};

class Solution {
public:
    vector<int> ans;

    vector<int> reversePrint(ListNode* head)
    {
        reverse(head);
        return ans;
    }

    void reverse(ListNode* head)
    {
        if (head == nullptr) {
            return;
        }

        reverse(head->next);
        ans.push_back(head->val);
    }
};